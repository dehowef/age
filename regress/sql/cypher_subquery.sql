LOAD 'age';
SET search_path TO ag_catalog;

SELECT * FROM create_graph('exists_subquery');

SELECT * FROM cypher('exists_subquery', $$ 
										CREATE (:person {name: "Briggite", age: 32})-[:knows]->(:person {name: "Takeshi", age: 28}),
											   (:person {name: "Faye", age: 25})-[:knows]->(:person {name: "Tony", age: 34})-[:loved]->(:person {name : "Valerie", age: 33}),
											   (:person {name: "Calvin", age: 6})-[:knows]->(:pet {name: "Hobbes"}),
											   (:person {name: "Charlie", age: 8})-[:knows]->(:pet {name : "Snoopy"})
										$$) AS (result agtype);

SELECT * FROM cypher('exists_subquery', $$ MATCH (a) RETURN (a) $$) AS (result agtype);

SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person) 
										   WHERE EXISTS {(a:person)-[]->(:pet)}
										   RETURN (a) $$) AS (result agtype);
-- trying to use b when not defined, should fail
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person) 
										   WHERE EXISTS {(a:person)-[]->(b:pet)}
										   RETURN (a) $$) AS (result agtype);
--query inside
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {MATCH (a:person)-[]->(b:pet) RETURN b}
										   RETURN (a) $$) AS (result agtype);

--repeat variable in match
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
						   				 			     MATCH (a:person)
						   				 			     WHERE a.name = 'Takeshi'
						   				 			     RETURN a
										   				 }
										   RETURN (a) $$) AS (result agtype);
--query inside, with WHERE
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {MATCH (a:person)-[]->(b:pet)
										   				 WHERE b.name = 'Briggite'
										   				 RETURN b}
										   RETURN (a) $$) AS (result agtype);


--no return
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {MATCH (a:person)-[]->(b:pet)
										   				 WHERE a.name = 'Calvin'}
										   RETURN (a) $$) AS (result agtype);

-- union
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (a:person)-[]->(b:pet)
										   				 WHERE b.name = 'Hobbes'
										   				 RETURN b
										   				 UNION
										   				 MATCH (c:person)-[]->(d:person)
										   				 RETURN c
										   				}
										   RETURN (a) $$) AS (result agtype);

-- union, mismatched var, should fail
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (a:person)-[]->(b:pet)
										   				 WHERE b.name = 'Snoopy'
										   				 RETURN c
										   				 UNION
										   				 MATCH (c:person)-[]->(d:person)
										   				 RETURN c
										   				 }
										   RETURN (a) $$) AS (result agtype);

--union, no returns
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (a:person)-[]->(b:pet)
										   				 WHERE a.name = 'Charlie'
										   				 UNION
										   				 MATCH (c:person)-[]->(d:person)
										   				}
										   RETURN (a) $$) AS (result agtype);

--union, mismatched returns, should fail
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (a:person)-[]->(b:pet)
										   				 WHERE a.name = 'Faye'
										   				 RETURN a
										   				 UNION
										   				 MATCH (c:person)-[]->(d:person)
										   				}
										   RETURN (a) $$) AS (result agtype);

--nesting
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE c.name = 'Takeshi'
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);

--nesting, accessing var in outer scope
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE b = c
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);

--nesting, accessing indirection in outer scope
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE b.name = 'Takeshi'
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);

--nesting, accessing var 2+ levels up
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE a.name = 'Takeshi'
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);

--nesting, accessing indirection 2+ levels up
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE a = b
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);

-- EXISTS outside of WHERE
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person) 
										   RETURN a, EXISTS {(a:person)-[]->(:pet)}
										   $$) AS (a agtype, exists agtype);

-- Var doesnt exist in outside scope, should fail
SELECT * FROM cypher('exists_subquery', $$ RETURN 1, 
										   		  EXISTS {
									   					  MATCH (b:person)-[]->(:pet)
									   					  RETURN a
									   					 }
										   $$) AS (a agtype, exists agtype);

--
-- Cleanup
--
SELECT * FROM drop_graph('exists_subquery', true);

--
-- End of tests
--
