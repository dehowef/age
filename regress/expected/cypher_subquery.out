LOAD 'age';
SET search_path TO ag_catalog;
SELECT * FROM create_graph('exists_subquery');
NOTICE:  graph "exists_subquery" has been created
 create_graph 
--------------
 
(1 row)

SELECT * FROM cypher('exists_subquery', $$ 
										CREATE (:person {name: "Briggite", age: 32})-[:knows]->(:person {name: "Takeshi", age: 28}),
											   (:person {name: "Faye", age: 25})-[:knows]->(:person {name: "Tony", age: 34})-[:loved]->(:person {name : "Valerie", age: 33}),
											   (:person {name: "Calvin", age: 6})-[:knows]->(:pet {name: "Hobbes"}),
											   (:person {name: "Charlie", age: 8})-[:knows]->(:pet {name : "Snoopy"})
										$$) AS (result agtype);
 result 
--------
(0 rows)

SELECT * FROM cypher('exists_subquery', $$ MATCH (a) RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
 {"id": 1688849860263937, "label": "pet", "properties": {"name": "Hobbes"}}::vertex
 {"id": 1688849860263938, "label": "pet", "properties": {"name": "Snoopy"}}::vertex
(9 rows)

SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person) 
										   WHERE EXISTS {(a:person)-[]->(:pet)}
										   RETURN (a) $$) AS (result agtype);
                                             result                                              
-------------------------------------------------------------------------------------------------
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(2 rows)

-- trying to use b when not defined, should fail
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person) 
										   WHERE EXISTS {(a:person)-[]->(b:pet)}
										   RETURN (a) $$) AS (result agtype);
ERROR:  variable `b` does not exist
LINE 2:              WHERE EXISTS {(a:person)-[]->(b:pet)}
                                                   ^
--query inside
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {MATCH (a:person)-[]->(b:pet) RETURN b}
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

--repeat variable in match
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
						   				 			     MATCH (a:person)
						   				 			     WHERE a.name = 'Takeshi'
						   				 			     RETURN a
										   				 }
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

--query inside, with WHERE
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {MATCH (a:person)-[]->(b:pet)
										   				 WHERE b.name = 'Briggite'
										   				 RETURN b}
										   RETURN (a) $$) AS (result agtype);
 result 
--------
(0 rows)

--no return
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {MATCH (a:person)-[]->(b:pet)
										   				 WHERE a.name = 'Calvin'}
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

-- union
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (a:person)-[]->(b:pet)
										   				 WHERE b.name = 'Hobbes'
										   				 RETURN b
										   				 UNION
										   				 MATCH (c:person)-[]->(d:person)
										   				 RETURN c
										   				}
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

-- union, mismatched var, should fail
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (a:person)-[]->(b:pet)
										   				 WHERE b.name = 'Snoopy'
										   				 RETURN c
										   				 UNION
										   				 MATCH (c:person)-[]->(d:person)
										   				 RETURN c
										   				 }
										   RETURN (a) $$) AS (result agtype);
ERROR:  could not find rte for c
LINE 5:                   RETURN c
                                 ^
--union, no returns
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (a:person)-[]->(b:pet)
										   				 WHERE a.name = 'Charlie'
										   				 UNION
										   				 MATCH (c:person)-[]->(d:person)
										   				}
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

--union, mismatched returns, should fail
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (a:person)-[]->(b:pet)
										   				 WHERE a.name = 'Faye'
										   				 RETURN a
										   				 UNION
										   				 MATCH (c:person)-[]->(d:person)
										   				}
										   RETURN (a) $$) AS (result agtype);
ERROR:  syntax error at or near "}"
LINE 8:                  }
                         ^
--nesting
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE c.name = 'Takeshi'
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

--nesting, accessing var in outer scope
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE b = c
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

--nesting, accessing indirection in outer scope
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE b.name = 'Takeshi'
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

--nesting, accessing var 2+ levels up
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE a.name = 'Takeshi'
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

--nesting, accessing indirection 2+ levels up
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
										   WHERE EXISTS {
										   				 MATCH (b:person)
										   				 WHERE EXISTS {
										   				 			   MATCH (c:person)
										   				 			   WHERE a = b
										   				 			   RETURN c
										   				 			  }
										   				 }
										   RETURN (a) $$) AS (result agtype);
                                              result                                               
---------------------------------------------------------------------------------------------------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex
(7 rows)

-- EXISTS outside of WHERE
SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person) 
										   RETURN a, EXISTS {(a:person)-[]->(:pet)}
										   $$) AS (a agtype, exists agtype);
                                                 a                                                 | exists 
---------------------------------------------------------------------------------------------------+--------
 {"id": 844424930131969, "label": "person", "properties": {"age": 32, "name": "Briggite"}}::vertex | false
 {"id": 844424930131970, "label": "person", "properties": {"age": 28, "name": "Takeshi"}}::vertex  | false
 {"id": 844424930131971, "label": "person", "properties": {"age": 25, "name": "Faye"}}::vertex     | false
 {"id": 844424930131972, "label": "person", "properties": {"age": 34, "name": "Tony"}}::vertex     | false
 {"id": 844424930131973, "label": "person", "properties": {"age": 33, "name": "Valerie"}}::vertex  | false
 {"id": 844424930131974, "label": "person", "properties": {"age": 6, "name": "Calvin"}}::vertex    | true
 {"id": 844424930131975, "label": "person", "properties": {"age": 8, "name": "Charlie"}}::vertex   | true
(7 rows)

-- Var doesnt exist in outside scope, should fail
SELECT * FROM cypher('exists_subquery', $$ RETURN 1, 
										   		  EXISTS {
									   					  MATCH (b:person)-[]->(:pet)
									   					  RETURN a
									   					 }
										   $$) AS (a agtype, exists agtype);
ERROR:  could not find rte for a
LINE 4:                    RETURN a
                                  ^
--
-- Cleanup
--
SELECT * FROM drop_graph('exists_subquery', true);
NOTICE:  drop cascades to 6 other objects
DETAIL:  drop cascades to table exists_subquery._ag_label_vertex
drop cascades to table exists_subquery._ag_label_edge
drop cascades to table exists_subquery.person
drop cascades to table exists_subquery.knows
drop cascades to table exists_subquery.loved
drop cascades to table exists_subquery.pet
NOTICE:  graph "exists_subquery" has been dropped
 drop_graph 
------------
 
(1 row)

--
-- End of tests
--
